src/generic/datavgen.cpp

```diff
@@ -2046,69 +2051,215 @@ bool wxDataViewMainWindow::EnableDropTarget( const wxDataFormat &format )
     return true;
 }
 
-void wxDataViewMainWindow::RemoveDropHint()
+void wxDataViewMainWindow::RemoveDropHint(bool refreshOnly/* = false*/)
 {
-    if (m_dropHint)
+    if (m_dropHint == 0)
+        return;
+
+    if (m_dropHint == 1)
+        RefreshRow(m_dropHintLine);
+    else if (m_dropHint == 2)
+        RefreshRows(m_dropHintLine == 0 ? 0 : m_dropHintLine - 1, m_dropHintLine);
+    else if (m_dropHint == 3)
+        RefreshRows(m_dropHintLine, std::max(m_dropHintLine + 1, GetRowCount() - 1));
+
+    if (!refreshOnly)
     {
-            m_dropHint = false;
-            RefreshRow( m_dropHintLine );
-            m_dropHintLine = (unsigned int) -1;
+        m_dropHint = 0;
+        m_dropHintIndentLevel = -1;
+        m_dropHintLine = (unsigned int) -1;
     }
 }
 
-wxDragResult wxDataViewMainWindow::OnDragOver( wxDataFormat format, wxCoord x,
-                                               wxCoord y, wxDragResult def )
+// Returns drop hint
+int wxDataViewMainWindow::GetDropItemInfo(const wxCoord x, const wxCoord y,
+                                          unsigned int* itemRow,
+                                          wxDataViewItem* item,
+                                          int* proposedDropIndex,
+                                          int* indentLevel)
 {
     int xx = x;
     int yy = y;
     m_owner->CalcUnscrolledPosition( xx, yy, &xx, &yy );
-    unsigned int row = GetLineAt( yy );
 
-    wxDataViewItem item;
+    unsigned int row = GetLineAt(yy);
+    *itemRow = row;     // [arg set]
+
+    if (row >= GetRowCount() || xx > GetEndOfLastCol())
+        return 0;
+
+    if (IsVirtualList())
+    {
+        *item = GetItemByRow(row);      // [arg set]
+
+        if (!item->IsOk())
+            return 0;
+    }
+    else    // TODO: DROP_INDEX_STYLE
```

Comment:
What does this comment mean?

Reply:
If we want to support an old-style of drop (when you can drop only inside and ProposedDropIndex == -1 always), we must extend the style enum of wxDVC with something like wxDV_PROPOSE_DROP_INDEX

Reply:
Do you mean for compatibility? Because I don't think we should worry about it here, the code which doesn't need the proposed drop index can just blithely ignore it...